docs/index.rst

```diff
@@ -714,6 +714,68 @@ method and ``Flask-Classful`` will take care of the rest::
             return "Looking at me? I guess you're logged in."
 
 
+Integrating with webargs
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the wider Flask ecosystem, `webargs <https://webargs.readthedocs.io>`_
+is a commonly-used library to parse and validate HTTP requests. There
+are a couple of things to bear in mind when using ``webargs`` with
+``Flask-Classful``.
+
+If using the `@use_args
+<https://webargs.readthedocs.io/en/latest/api.html#webargs.core.Parser.use_args>`_
+decorator, the parsed arguments are injected into your view. However,
+by default, the args variable would also be included in your route.
+e.g.::
+
+    class UsersView(FlaskView):
+        @use_args(...)
+        def post(self, id, args):
+           # the rest of the view
+
+will result in a route such as ``/users/<id>/<args/`` which is obviously
+not what you want. The workaround is to include ``base_args = ['args`]``
+in your view, which causes `build_rule <http://flask-classful.teracy.org/#api>`_
+to ignore that argument.
+
+Next, if you are using `marshmallow
+<https://marshmallow.readthedocs.io/en/3.0/>`_ and specifically `schema factories
+<https://webargs.readthedocs.io/en/latest/advanced.html#schema-factories>`_,
+by default ``Flask-Classful`` will attempt to create endpoints from the
+functions, which crashes and burns. This was discussed above in the
+`Hiding your own methods (they're not all special!)`_ section, yet it
+may not be obvious that this is the failure at first glance.
+
+The first workaround applies if you're using a vanilla schema factory
+function::
+
+    from marshmallow import Schema, fields
+    from webargs.flaskparser import use_args
+
+    class UserSchema(Schema):
+        # the rest of the schema
+
+    def make_user_schema(request):
+        # Filter based on 'fields' query parameter
+        only = request.args.get("fields", None)
+        # Respect partial updates for PATCH requests
+        partial = request.method == "PATCH"
+        # Add current request to the schema's context
+        return UserSchema(only=only, partial=partial, context={"request": request})
+
+    class UsersView(FlaskView):
+        excluded_methods = ['make_user_schema']  # Important
+        @use_args(make_user_schema)
```

Comment:
did not work to me 
```python
  UsersView.register(app)
  File "/opt/app/flask_classful.py", line 196, in register
    .format(name, cls.__name__))
flask_classful.DecoratorCompatibilityError: Incompatible decorator detected on post in class UsersView
```

Reply:
this works to me (by using `base_args = ['args']` and `@route` decorator https://github.com/teracyhq/flask-classful/blob/develop/test_classful/test_base_args.py#L65

Reply:
if you can add more tests (as examples) to support the docs, it would be great, otherwise, seems that it's not working to me.

Reply:
@hoatle I think (and hope!) you're running into what I fixed in https://github.com/teracyhq/flask-classful/pull/106/ (still up in the air). Can you try again with the fix in https://github.com/teracyhq/flask-classful/pull/106/commits/34afd39a883abf4f25c5523670b55d1864f541e8 and let me know if it works?

Reply:
@antgel I merged your PR and added unit tests https://github.com/teracyhq/flask-classful/commit/a046114e9e3bface851fbee1d833ef3da5cfec43

From the unit tests, I don't see why we need to use `excluded_methods`. Using `base_args = ['args']` is enough to get the function view work with webargs.

Reply:
Interesting point, thanks for the feedback, I'll take a look at that.